File: python/java_to_python/05-pgmAbstraccion/src/co/edu/campusucc/poo/TipoCombustion.py
from Vehiculo import Vehiculo
import time


class TipoCombustion(Vehiculo):

    def __init__(self):
        super().__init__()
        self.setTipo("Combustion")

    def startingEngine(self):
        print("⛽Check Fuel...✅")
        for i in range(3):
            print("💥 Spark...💥")
            time.sleep(1 - (i * 0.2))
        print("Started...🚘🚘")

    def stopingEngine(self):
        print("🚗Check Engine ✅")
        for i in range(2):
            print("⛔ Stoping Engine...⛔‼️")
            time.sleep(1 - (i * 0.4))
        print("Stoped...⛔")

File: ruby/pgms_ruby/05-pgmAbstraccion/TipoCombustion.rb
require_relative 'Vehiculo'
require 'time'

class TipoCombustion < Vehiculo
  def initialize
    super
    setTipo("Combustion")
  end

  def startingEngine
    puts "⛽Check Fuel...✅"
    3.times do |i|
      puts "💥 Spark...💥"
      sleep(1 - (i * 0.2))
    end
    puts "Started...🚘🚘"
  end

  def stopingEngine
    puts "🚗Check Engine ✅"
    2.times do |i|
      puts "⛔ Stoping Engine...⛔‼️"
      sleep(1 - (i * 0.4))
    end
    puts "Stoped...⛔"
  end
end